# ==============================================================================
# FIXING COMMIT DIFF
# ==============================================================================
# File: widget/gtk/WaylandSurface.h
# Commit: 16d572654502
# Full Hash: 16d57265450277b886bc0c6affccb906ff511f14
# Author: stransky <stransky@redhat.com>
# Date: 2024-12-27 21:21:05
# Description:
#   Bug 1938475 [Wayland] Fallback to monitor screen scale if we're missing wayland surface one r=emilio
#   
#   Differential Revision: https://phabricator.services.mozilla.com/D232705
# ==============================================================================

diff -r 33e8d495b853 -r 16d572654502 widget/gtk/WaylandSurface.h
--- a/widget/gtk/WaylandSurface.h	Fri Dec 27 12:19:24 2024 +0000
+++ b/widget/gtk/WaylandSurface.h	Fri Dec 27 14:16:37 2024 +0000
@@ -175,6 +175,10 @@
   static constexpr const double sNoScale = -1;
   double GetScale();
 
+  // The same as GetScale() but returns monitor scale if window scale is
+  // missing.
+  double GetScaleSafe();
+
   // Called when screen ceiled scale changed or set initial scale before we map
   // and paint the surface.
   void SetCeiledScaleLocked(const WaylandSurfaceLock& aProofOfLock,
